openeo_udp/fusets_mogpr/set_path.py

```diff
@@ -0,0 +1,92 @@
+#%%
+
+import os
+import sys
+import zipfile
+import requests
+import tempfile
+import shutil
+import functools
+
+from openeo.udf import inspect
+
+def download_file(url, path):
+    """
+    Downloads a file from the given URL to the specified path.
+    """
+    response = requests.get(url, stream=True)
+    with open(path, "wb") as file:
+        file.write(response.content)
+
+def extract_zip_to_temp(zip_path):
+    """
+    Extracts a zip file to a temporary directory.
+    """
+    # Create a temporary directory
+    temp_dir = tempfile.mkdtemp()
+
+    # Extract the zip file to the temporary directory
+    with zipfile.ZipFile(zip_path, "r") as zip_ref:
+        zip_ref.extractall(temp_dir)
+
+    return temp_dir
+
+def move_top_level_folder_to_destination(temp_dir, destination_dir):
+    """
+    Moves the first top-level folder from the temporary directory to the destination directory.
```

Comment:
doc states "first top-level folder", but in implementation I see a for loop, so that means "each top level folder"